bjerk-style.md

```diff
@@ -0,0 +1,411 @@
+  for a decision, it not only increases the hearer's understanding, and makes them more likely to
+  adhere or comply, but it also shares criteria with them with which to evaluate the decision and
+  its importance.
+
+* **Explicitly pass options to library functions at the call site, instead of relying on the
+   defaults**. For example, write `@prefetch(a, .{ .cache = .data, .rw = .read, .locality = 3 });`
+   over `@prefetch(a, .{});`. This improves readability but most of all avoids latent,
+   potentially catastrophic bugs in case the library ever changes its defaults.
+
+
+## Performance
+
+> “The lack of back-of-the-envelope performance sketches is the root of all evil.”
+> — Rivacindela Hudsoni
+
+* Think about performance from the outset, from the beginning. **The best time to solve performance,
+  to get the huge 1000x wins, is in the design phase, which is precisely when we can't measure or
+  profile.** It's also typically harder to fix a system after implementation and profiling, and the
+  gains are less. So you have to have mechanical sympathy. Like a carpenter, work with the grain.
+
+* **Perform back-of-the-envelope sketches with respect to the four resources (network, disk, memory,
+  CPU) and their two main characteristics (bandwidth, latency).** Sketches are cheap. Use sketches
+  to be “roughly right” and land within 90% of the global maximum.
+
+* Optimize for the slowest resources first (network, disk, memory, CPU) in that order, after
+  compensating for the frequency of usage, because faster resources may be used many times more. For
+  example, a memory cache miss may be as expensive as a disk fsync, if it happens many times more.
+
+* Distinguish between the control plane and data plane. A clear delineation between control plane
+  and data plane through the use of batching enables a high level of assertion safety without losing
+  performance. See our [July 2021 talk on Zig SHOWTIME](https://youtu.be/BH2jvJ74npM?t=1958) for
+  examples.
+
+* Amortize network, disk, memory and CPU costs by batching accesses.
+
+* Let the CPU be a sprinter doing the 100m. Be predictable. Don't force the CPU to zig zag and
+  change lanes. Give the CPU large enough chunks of work. This comes back to batching.
+
+* Be explicit. Minimize dependence on the compiler to do the right thing for you.
+
+## Developer experience
+
+> “There are only two hard things in Computer Science: cache invalidation, naming things, and
+> off-by-one errors.”
+> — Phil Karlton
+
+### Naming things
+
+* **Get the nouns and verbs just right.** Great names are the essence of great code, they capture
+  what a thing is or does, and provide a crisp, intuitive mental model. They show that you
+  understand the domain. Take time to find the perfect name, to find nouns and verbs that work
+  together, so that the whole is greater than the sum of its parts.
+
+* Use `snake_case` for function, variable, and file names. The underscore is the closest thing we
```

Comment:
our standard has been camelCasing for variables and functions, are we going to step away from that pattern?